src/rgw/rgw_lua.h

```diff
@@ -34,5 +34,22 @@ int read_script(rgw::sal::RGWRadosStore* store, const std::string& tenant, optio
 // delete the stored lua script from a context
 int delete_script(rgw::sal::RGWRadosStore* store, const std::string& tenant, optional_yield y, context ctx);
 
+#ifdef WITH_RADOSGW_LUA_MODULES
+#include <set>
+
+using modules_t = std::set<std::string>;
+
+const std::string LUA_MODULES_INSTALL_DIR = "./.luarocks";
```

Comment:
should this be located somewhere under `/var/lib` instead of hiding it in a dot directory ?

Reply:
but how would that work in case of a vstart cluster?
in theory, this is just a cache, as the needed lua rocks are installed on every restart of the RGW.
maybe we should put that under `/tmp`? or based on an env var?

Reply:
> but how would that work in case of a vstart cluster?

we have the same issue when it comes to asok, log files and more. we can address this problem using a dedicated option. i don't think `/tmp` is the right place to cache lua modules. according to [FHS](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s08.html), `/var/lib` is the right place.

Reply:
ok. according to this: https://github.com/ceph/ceph/blob/master/src/vstart.sh#L1526
I will have to add that as a parameter to the rgw executable

Reply:
> > but how would that work in case of a vstart cluster?
> 
> we have the same issue when it comes to asok, log files and more. we can address this problem using a dedicated option. i don't think `/tmp` is the right place to cache lua modules. according to [FHS](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s08.html), `/var/lib` is the right place.

one of the intents in using only a local directory was to be separated from the environment of the host
- rocks installed on the host cannot be used by rgw if not in the allowlist
- rocks installed via the allowlist are not added to the host (this is also for the case that rgw is not allowed to change system directories)
- BTW, the official locations are:
 - `/usr/share/lua/` for the lua code of the rocks
- `/usr/lib/lua/` for .so files of the rocks
- `/usr/lib/luarocks/` for metadata

Reply:
make the directory configurable. by default it is set to `/tmp/luarocks` so it will be separated from any lua rocks installed on the host. but could be set to another location